### DISCUSSION TITLE
Parallelizing rdkit.Chem.DataStruct.BulkTanimotoSimilarity

### DISCUSSION THREAD

**Discussion by ghost** [rdkit/rdkit#3999] [Q&A]
I have two very long lists of mols `query_mols` and `target_mols` for which i want to calculate the pairwise similarity. I have found that iterating over `query_mols` and running  `BulkTanimotoSimilarity(query_mols[i], target_mols)` is much quicker than  using two for loops with `TanimotoSimilarity(query_mols[i], target_mols[j])`. I thought that i could speed this up even further by parallelizing over `query_mols` with python multiprocessing. However, instead of a speed up, I'm seeing a slowdown of 10-20x over just iterating over `query_mols`. Seems like there's some kind of shared resource for which the processes might be competing? Has anyone ever tried this before? Any ideas on what is causing/How to resolve?

**Comment by greglandrum** [comment]
@beazer001 : without seeing code it's hard to figure out what's going on here.
There are no shared resources used in the similarity calculation.
Calling TanimotoSimilarity() repeatedly, whether parallel or not, is going to be really slow compared to calling BulkTanimotoSimilarity due to the number of times you have to cross the Python<->C++ border.
